dbt/tests/generic/test_res_class_matches_pardat.sql

```diff
@@ -3,17 +3,31 @@
 -- filters for residential parcels by anti-joining the model against `comdat`
 -- using parid and taxyr; as a result, it filters out mixed-use parcels as well.
 --
--- By default, the test will compare the first 3 digits of each set of classes;
--- if `major_class_only=true`, however, the test will compare the first digit
--- only.
+-- Optional parameters:
+--
+-- * major_class_only (bool): Compare only the first digit of classes. When
+-- set to False, compare the first three digits instead.
```

Comment:
**issue (non-blocking):** We should make sure that these checks work for incentive (e.g. `597A`) and omitted classes (e.g. `OA2`).

Reply:
Let me break my follow-up questions down by value of `major_class_only`:

1. When `major_class_only=False`, omitted classes should be fine as long as they're all 3 digits and they need to match exactly (i.e. `OA2` is not equivalent to `OA5`). Do incentive classes require a match in the letter suffix? If so, perhaps we need to compare four digits instead of three.
2. When `major_class_only=True`, do we want all omitted classes to match each other (i.e. `OA2` is equivalent to `OA5`), or do we need a special case for those? And is it fine to still compare the first digit of incentive classes?

Reply:
1. I don't know enough about them yet to say definitively, but for now let's assume they need to match on all 3 characters.
2. Annoyingly, we'll probably have to build in logic for a special case, since the major class for the omitted is the _last_ digit.

Reply:
> Annoyingly, we'll probably have to build in logic for a special case, since the major class for the omitted is the last digit.

So to clarify, does this mean that e.g. a PIN with class `OA2` should have a pardat class of either `OA2` or `2xx`? I just spot checked the current failures for omitted classes, and all of the mismatched classes are `OA2` or `OA5` matching to either `EX` or `100` (no clear pattern I could discern from a cursory glance based on class alone). Is it possible that those are valid values for omitted classes?

Reply:
The OA classes are sufficiently new that I actually don't know. I'll bring it up tomorrow with MV. Feel free to merge + fast-follow if you don't want this blocked.

Reply:
So from what I understand, the `OA2`, `OA3`, and `OA5` codes should only exist in the various detail tables (`COMDAT`, `DWELDAT`, etc.). If they exist they can basically have _any_ class code in `pardat`, but should match `EX`, `100`, or their corresponding major class most of the time. Further, if something has an `OA` class code, it should have a corresponding `asmt.tot57` value in the same tax year, so we should add a test for that.

Reply:
Issue open here: https://github.com/ccao-data/data-architecture/issues/363